### PR TITLE
docs(samples): add list jobs

### PR DESCRIPTION
**Pull request by pmakani**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [X] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-bigquery/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [X] Ensure the tests and linter pass
- [X] Code coverage does not decrease (if any source code was changed)
- [X] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: samples/snippets/src/test/java/com/example/bigquery/ListJobsIT.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  @After
+  public void tearDown() {
+    // Clean up
+    System.setOut(null);
+  }
+
+  @Test
+  public void testListJobs() {
+    ListJobs.listJobs();
+    Truth.assertThat(bout.toString()).contains("Success! Job ID");
```
Comment (stephaniewang526) [inline samples/snippets/src/test/java/com/example/bigquery/ListJobsIT.java:59]: I dont think we need `Truth.`

File: samples/snippets/src/main/java/com/example/bigquery/ListJobs.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.BigQueryException;
+import com.google.cloud.bigquery.BigQueryOptions;
+import com.google.cloud.bigquery.Job;
+
+// Sample to get list of jobs
+public class ListJobs {
+
+  public static void runListJobs() {
+    listJobs();
```
Comment (stephaniewang526) [inline samples/snippets/src/main/java/com/example/bigquery/ListJobs.java:30]: Do we not need to pass in dataset for listing jobs?

File: samples/snippets/src/test/java/com/example/bigquery/ListJobsIT.java (hunk 1 above)
Comment (pmakani) [reply samples/snippets/src/test/java/com/example/bigquery/ListJobsIT.java:59]: Done

File: samples/snippets/src/main/java/com/example/bigquery/ListJobs.java (hunk 2 above)
Comment (pmakani) [reply samples/snippets/src/main/java/com/example/bigquery/ListJobs.java:30]: there is no option to pass dataset.

**Pull request merged**